### PR TITLE
Pyband: Declared the error to pyband world

### PR DESCRIPTION
**Pull request by pzshine**
## Implementation details

- Declared the custom error to pyband

### Here is the error list

- EmptyRequestMsgError
- QueryError
- NegativeIntegerError
- ValueTooLargeError
- InsufficientCoinError
- EmptyMsgError
- NotFoundError
- UndefinedError
- DecodeError
- ConvertError
- UnsuccessfulCallError
- CreateError
- SchemaError  

## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [x] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: helpers/pyband/pyband/error.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
+class EmptyRequestMsgError(Exception):
```
Comment (taobun) [inline helpers/pyband/pyband/error.py]: I think renaming this file to `exceptions.py` sounds better.

File: helpers/pyband/pyband/error.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    def __init__(self, message: str):
+        super().__init__(message)
+
+
+class QueryError(Exception):
+    def __init__(self, message: str):
```
Comment (taobun) [inline helpers/pyband/pyband/error.py]: I think use `pass` in case we don't want to modify any input.

**Review by taobun — CHANGES_REQUESTED** [review]

File: helpers/pyband/pyband/exceptions.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
+class EmptyRequestMsgError(Exception):
```
Comment (taobun) [inline helpers/pyband/pyband/exceptions.py]: Swit recommend to have
```
class PybandError(Exception):
    pass

class EmptyRequestMsgError(PybandError):
    pass
```

File: helpers/pyband/pyband/data.py [hunk 4]
```diff
@@ -24,7 +25,7 @@ def as_json(self) -> dict:
 
     def validate(self) -> bool:
         if self.amount < 0:
-            raise ValueError("Expect amount more than 0")
+            raise InsufficientCoinError("Expect amount more than 0")
```
Comment (taobun) [inline helpers/pyband/pyband/data.py]: I think it can use ValueError or NegativeIntegerError (a coin cannot be less than zero)

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by taobun — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**